src/components/SearchBar/index.jsx

```diff
@@ -0,0 +1,93 @@
+import "./index.css";
+
+import React, { useEffect, useState, memo } from "react";
+import { auth, db } from "../../lib/firebase";
+
+import { Box } from "@mui/material";
+import Post from "../Post";
+import ShareModal from "../../reusableComponents";
+import SideBar from "../SideBar";
+
+const MemoizedPost = memo(Post);
+
+function SearchBar() {
+  const [searchText, setSearchText] = useState("");
+  const [openShareModal, setOpenShareModal] = useState(false);
+  const [currentPostLink, setCurrentPostLink] = useState("");
+  const [postText, setPostText] = useState("");
+  const [posts, setPosts] = useState([]);
+
+  const handleSearch = (e) => {
+    setSearchText(e.target.value);
+  };
+
+  // Code to fetch posts from database
+
+  useEffect(() => {
+    const fetchPosts = async () => {
+      db.collection("posts")
+        .orderBy("timestamp", "desc")
```

Comment:
hello @BhartiNagpure , fetching the whole db and then searching on client side is not practical as it can be seen in [this link](https://firebase.google.com/docs/firestore/solutions/search).